vertx-pojo-mapper-common-test/src/main/java/de/braintags/vertx/jomnigate/testdatastore/mapper/typehandler/BaseRecord.java

```diff
@@ -241,6 +243,20 @@ private boolean compareNumber(Object value, Object compareValue, String fieldNam
   private boolean compareArray(Object value, Object compareValue, String fieldName) {
     if (!compareValue.getClass().isArray())
       throw new MappingException("Contents are not equal: " + fieldName);
+    int l1 = Array.getLength(value);
+    int l2 = Array.getLength(compareValue);
+    if (l1 != l2) {
+      throw new MappingException("different lenght in arrays: " + l1 + " / " + l2);
+    }
+    // first clear wether all elements are existing, equal the order
```

Comment:
This part is only to reproduce the existing ability to search by name. Additionally will come #79 as refactoring job, where inside we can define the correct behaviour of both systems, where an administratable factory system should be defined.